src/client.ts

```diff
@@ -685,7 +685,7 @@ export class BareClient {
     }
 
     if (shouldReturnCachedData(cachePolicy)) {
-      onData(entryState);
+      setTimeout(() => onData(entryState));
```

Comment:
I guess a microtrask is more appropriate here
```suggestion
      Promise.resolve().then(() => onData(entryState));
```

See https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/